### PR TITLE
[eclipse/xtext-eclipse#809] Cache task marker pattern for performance

### PR DESCRIPTION
**Pull request by ArneDeutsch**
Signed-off-by: Arne Deutsch <Arne.Deutsch@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext/src/org/eclipse/xtext/tasks/DefaultTaskParser.xtend [hunk 1]
```diff
@@ -8,15 +8,17 @@
...
 import org.eclipse.xtext.util.Strings
-import java.util.Map
 
 /**
  * @author Stefan Oehme - Initial contribution and API
  * @since 2.6
  */
 class DefaultTaskParser implements ITaskParser {
+	
+	var patternCache = new HashMap<TaskTags, Pattern>
```
Comment (kthoms) [inline org.eclipse.xtext/src/org/eclipse/xtext/tasks/DefaultTaskParser.xtend]: could use val here

File: org.eclipse.xtext/src/org/eclipse/xtext/tasks/DefaultTaskParser.xtend [hunk 2]
```diff
@@ -60,11 +62,16 @@ class DefaultTaskParser implements ITaskParser {
 	}
 
 	protected def toPattern(TaskTags taskTags) {
```
Comment (kthoms) [inline org.eclipse.xtext/src/org/eclipse/xtext/tasks/DefaultTaskParser.xtend]: Move Pattern into TaskTags as field, would make cache obsolete

Comment (ArneDeutsch) [reply org.eclipse.xtext/src/org/eclipse/xtext/tasks/DefaultTaskParser.xtend]: Done.

File: org.eclipse.xtext/xtend-gen/org/eclipse/xtext/tasks/DefaultTaskParser.java [hunk 3]
```diff
@@ -39,7 +37,7 @@
         return Collections.<Task>unmodifiableList(CollectionLiterals.<Task>newArrayList());
       }
       final Map<String, TaskTag> taskTagsByName = this.getTaskTagsByName(taskTags);
-      final Matcher matcher = this.toPattern(taskTags).matcher(source);
+      final Matcher matcher = taskTags.toPattern().matcher(source);
```
Comment (szarnekow) [inline org.eclipse.xtext/xtend-gen/org/eclipse/xtext/tasks/DefaultTaskParser.java]: If we do no longer call toPattern, we do not need to keep that method.

**Pull request merged**